livox_ros2_driver/livox_ros2_driver/lds_lidar.cpp

```diff
@@ -244,11 +244,13 @@ void LdsLidar::OnDeviceChange(const DeviceInfo *info, DeviceEvent type) {
     if (p_lidar->connect_state == kConnectStateOff) {
       p_lidar->connect_state = kConnectStateOn;
       p_lidar->info = *info;
-      g_lds_ldiar->connected_lidars_[p_lidar->info.broadcast_code] = &p_lidar->info;
+      // g_lds_ldiar->connected_lidars_[p_lidar->info.broadcast_code] = &p_lidar->info;
+      g_lds_ldiar->connected_lidars_[p_lidar->info.broadcast_code] = p_lidar->info;
```

Comment:
What happens if you copy DeviceInfo and print it here?

Reply:
And print original data and compare them.

Reply:
sorry how to copy?
Just print ?
```
      std::cout << g_lds_ldiar->connected_lidars_[p_lidar->info.broadcast_code].status.status_code.lidar_error_code.temp_status
                << ", "
                << p_lidar->info.status.status_code.lidar_error_code.temp_status
                << std::endl;
```